src/components/pagination/demo/mock-pagination-data.ts

```diff
@@ -0,0 +1,54 @@
+/**
+ * These functions mimic the structure of WordPress's pagination objects.
+ *
+ * @see https://developer.wordpress.org/reference/functions/paginate_links/
+ * @see https://timber.github.io/docs/guides/pagination/
+ */
+
+function mockPageData(page = 1, current = 1) {
+  return {
+    title: page,
+    text: page,
+    name: page,
+    current: page === current,
+    link: page !== current && `#page-${page}`,
+  };
+}
+
+export default function mockPaginationData({
+  current = 1,
+  midSize = 2,
+  total = 36,
+} = {}) {
+  const prev = current > 1 && mockPageData(current - 1, current);
+  const next = current < total && mockPageData(current + 1, current);
+  const pages: object[] = [];
+  let start = current - midSize;
+  let end = current + midSize;
+
+  if (start < 1) {
+    end += 1 - start;
+  }
+
+  if (end > total) {
+    start -= end - total;
+  }
+
+  start = Math.max(1, start);
```

Comment:
I _think_ what you're doing here is saying "take the largest value of either `1` or the `start` value", but a comment would be helpful. Perhaps I'm overly sensitive to this as a newer JS dev, but it's helpful to comment your code with the assumption that a junior developer with less familiarity with the codebase will end up maintaining this code.

I've never been frustrated to find _too many_ comments, but I'm often frustrated by the lack of them.

Reply:
- [x] Added line-by-line comments to this function